symengine/tests/basic/test_basic.cpp

```diff
@@ -74,20 +88,151 @@ TEST_CASE("Symbol hash: Basic", "[basic]")
 
 TEST_CASE("Symbol dict: Basic", "[basic]")
 {
-    umap_basic_num d;
+    umap_basic_num ubn;
+    map_basic_num mbn;
+    umap_basic_basic ubb;
+    map_basic_basic mbb;
+    vec_basic vb;
+    set_basic sb;
     RCP<const Basic> x = symbol("x");
     RCP<const Basic> x2 = symbol("x");
     RCP<const Basic> y = symbol("y");
+    RCP<const Number> i2 = integer(2);
+    RCP<const Number> i3 = integer(3);
     bool p = (x != x2);
     REQUIRE(p);           // The instances are different...
     REQUIRE(eq(*x, *x2)); // ...but equal in the SymPy sense
 
-    insert(d, x, integer(2));
-    insert(d, y, integer(3));
-
-    // Test printing:
-    std::cout << d << std::endl;
-    std::cout << *x << std::endl;
+    std::stringstream buffer;
+    buffer << ubn;
+    REQUIRE(buffer.str() == "{}");
+    insert(ubn, x, i2);
+    buffer.str("");
+    buffer << ubn;
+    REQUIRE(buffer.str() == "{x: 2}");
+    insert(ubn, y, i3);
+    buffer.str("");
+    buffer << mbn;
+    REQUIRE(buffer.str() == "{}");
+    insert(mbn, x, i2);
+    insert(mbn, y, i3);
+    buffer.str("");
+    buffer << mbb;
+    REQUIRE(buffer.str() == "{}");
+    insert(mbb, x, i2);
+    insert(mbb, y, i3);
+    buffer.str("");
+    buffer << ubb;
+    REQUIRE(buffer.str() == "{}");
+    insert(ubb, x, i2);
+    insert(ubb, y, i3);
+    buffer.str("");
+    buffer << vb;
+    REQUIRE(buffer.str() == "[]");
+    vb.push_back(x);
+    vb.push_back(i3);
+    buffer.str("");
+    buffer << vb;
+    REQUIRE(buffer.str() == "[x, 3]");
+    REQUIRE(vec_basic_eq(vb, {x, i3}));
+    REQUIRE(vec_basic_compare(vb, {x, i3}) == 0);
+    REQUIRE(not vec_basic_eq(vb, {i3, x}));
+    REQUIRE(vec_basic_eq_perm(vb, {i3, x}));
+    REQUIRE(not vec_basic_eq(vb, {i3}));
+    REQUIRE(not vec_basic_eq_perm(vb, {i3}));
+    REQUIRE(vec_basic_compare(vb, {i3}) == 1);
+    buffer.str("");
+    buffer << sb;
+    REQUIRE(buffer.str() == "[]");
+    sb.insert(i2);
+    sb.insert(y);
+
+    auto check_map_str =
+        [](std::string to_chk, std::vector<std::string> key, std::vector<std::string> val)
+    {
+        if(key.size() != val.size())
+            return false;
+        unsigned pos;
+        for(unsigned i = 0; i < key.size(); i++)
+        {
+            if (to_chk.substr(to_chk.find(key[i]) + 1, 3) != std::string(": " + val[i]))
+                return false;
+        }
+        return true;
+    };
+
+    buffer.str("");
+    buffer << ubn;
+    REQUIRE(check_map_str(buffer.str(), {"x", "y"}, {"2", "3"}));
```

Comment:
Well the algorithm kind of does the same.
It searches for occurrence of `x` and then checks that next three characters make `: 2`.